### PR TITLE
Climb

### LINKED ISSUE #11: Check solenoid addressing and test the climber mechanism

### PR DESCRIPTION
**Pull request by KaiFrc**
# Purpose: Created climb functionality for the robot. Tested and climbs as intended to the middle rung.
## Files Changed...

Climb.java: Climb Subsystem
RetractClimb.java: Command that retracts solenoid
ExtendClimb.java: Command that extends solenoid
Constants.java:  Added necessary pneumatic/climber constants for easier readability
RobotContainer.java: Controller info

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -70,18 +74,32 @@ public static RobotContainer getInstance() {
...
         XboxController.Button.kRightBumper.value);
     xboxFeedBtn.whileHeld(new Fire(m_feeder, m_launcher));
-
+    
+    // Tracks target
     final JoystickButton xboxTargetBtn = new JoystickButton(m_operatorXboxController,
         XboxController.Button.kLeftBumper.value);
     xboxTargetBtn.whileHeld(new TrackTarget(m_drivetrain, m_targeting));
-
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java:89]: Extra whitespace

File: src/main/java/frc/robot/commands/ExtendClimb.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_SETTING
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_SETTING
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=REQUIRES
+
+        m_climb = subsystem;
+        addRequirements(m_climb);
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/ExtendClimb.java]: Extra whitespace. We can also remove the auto generated comments

File: src/main/java/frc/robot/commands/ExtendClimb.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        return false;
+    }
+
+    @Override
+    public boolean runsWhenDisabled() 
+    {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+        return false;
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/ExtendClimb.java]: Extra whitespace. We can also remove the auto generated comments

File: src/main/java/frc/robot/commands/RetractClimb.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+// END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=IMPORTS
+
+/**
+ *
+ */
+public class RetractClimb extends CommandBase 
+{
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_DECLARATIONS
+        private final Climb m_climb;
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/RetractClimb.java:25]: Extra whitespace. We can also remove the auto generated comments

Comment (bjmcternan) [reply src/main/java/frc/robot/commands/RetractClimb.java:25]: In fact, this file needs to be cleaned up

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+// END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=IMPORTS
+
+/**
+ *
+ */
+
+public class Climb extends SubsystemBase 
+{
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTANTS
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Climb.java]: Cleanup this file. Remove extra whitespace and auto generated comments

**Review by bjmcternan — CHANGES_REQUESTED** [review]
This PR looks OK functionally, but it needs to be cleaned up

**Comment by KaiFrc** [comment]
We're using one double solenoid instead of two now, same. cleaned the code.

**Review by bjmcternan — CHANGES_REQUESTED** [review]
Let's merge main into this first. The constants file are going to have conflicts.

**Comment by KaiFrc** [comment]
I merged the main file into climb yesterday. It should go through

**Comment by bjmcternan** [comment]
There's stuff on the other PRs that is about to go in which will cause conflicts. 

1. Launcher PR
2. Emoji removal PR
3. This PR

**Comment by KaiFrc** [comment]
Do you suggest we merge the other PRS first and then merge this PR into main?

**Comment by KaiFrc** [comment]
I successfully merged the other PRS and resolved the merging fixes, awaiting approval.

**Comment by bjmcternan** [comment]
Oops one sec

**Comment by KaiFrc** [comment]
Accidentily closed instead of comment MB

**Review by bjmcternan — APPROVED** [review]
Don't break the launcher. That PR needs to go in now

**Review by bjmcternan — CHANGES_REQUESTED** [review]
Hold on this changes constants that cannot change

**Pull request merged**